### ISSUE TITLE
automatic check for spelling consistency?

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-doc#319]
For terms such as:
-  `datetime` vs `date-time` vs `date time`
- inter-cycle vs intercycle
- cycle-point vs cyclepoint
- etc.

Sphinx extension, or post commit hook, or simple check script in CI ?

**Comment by oliver-sanders** [comment]
https://github.com/sphinx-contrib/spelling

**Comment by wxtim** [comment]
Recently I tried to implement this and found it a bit tricky - sphinx-contrib/spelling isn't very well supported. Worse its lexers don't recognize apostrophe contractions so you have to exclude "don", "aren" &c &c.

One possibility is to just use it anyway and use sed or similar to extract all the problem words and add them to one or more sphixcotrib-spellings custom dictionaries (which we can configure). This will at least catch new errors.

**Comment by hjoliver** [comment]
> Worse its lexers don't recognize apostrophe contractions so you have to exclude "don", "aren" &c &c.

Yikes, that is indeed "worse".

**Comment by wxtim** [comment]
RE: Just using sphinxcontrib-spelling
I tried opening an issue https://github.com/sphinx-contrib/spelling/issues/146 on sphinxcontrib spelling - but it turns out that the problem is with the upstream pyenchant's tokenizer (see https://github.com/sphinx-contrib/spelling/issues/126).
If we did decided to live with the issues, one possible solution is to have separate dictionaries to allow us to better track why words are allowed: I'd propose
- spelling/dictionary.txt - words we definitely want to add
- spelling/not-real-words.txt - Definitely not real words, but we want them anyway: e.g. "foo", "bar" &c
- spelling/spellcheck.txt - "words" where there is some sort of lexer problem 
- spelling/TODO.txt - words which are contentious in some way or where spelling should be standardised at some point in the future - examples including any word ending ise/ize (which ought to be consistent) and Parameterize and Parametrize, which as far as I can tell are merely alternate spellings.

Another option is to write an entirely separate check using Ispell or Aspell. I couldn't do it in the 15 minutes I spent trying, which hardly rules it out.
